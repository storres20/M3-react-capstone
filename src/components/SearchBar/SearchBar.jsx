/* eslint-disable max-len */
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import styles from './SearchBar.module.css';
// Redux
import { updateMovie } from '../../redux/movies/movies';
// ***

function SearchBar() {
  /* Load Redux State */
  const allData = useSelector((state) => state.allData);

  /* Setup Redux dispatch */
  const dispatch = useDispatch();

  /* allData = movies (original values) */

  // Filter search bar
  const handleSearch = (event) => {
    const keyword = event.target.value;

    if (keyword !== '') {
      const results = allData.filter((user) => user.title.toLowerCase().includes(keyword.toLowerCase()));

      dispatch(updateMovie(results));
    } else {
      dispatch(updateMovie(allData));
    // If the text field is empty, show all users
    }
  };
  return (
    <div>
      <div className={styles.divFlex}>
        <input className={styles.inputFlex} type="text" placeholder="Search by title..." onChange={(event) => handleSearch(event)} />
      </div>
    </div>
  );
}

export default SearchBar;
