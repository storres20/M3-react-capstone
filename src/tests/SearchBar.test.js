import React from 'react';
import renderer from 'react-test-renderer';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import SearchBar from '../components/SearchBar/SearchBar';
import moviesReducer from '../redux/movies/movies';

// configureStore
const createTestStore = () => {
  const store = configureStore({
    reducer: {
      movie: moviesReducer,
    },
  });

  return store;
};
export default createTestStore;

// Test
describe('My Connected React-Redux Component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = createTestStore();

    // Create a redux store
    component = renderer.create(
      <Provider store={store}>
        <SearchBar />
      </Provider>,
    ).toJSON();
  });

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
