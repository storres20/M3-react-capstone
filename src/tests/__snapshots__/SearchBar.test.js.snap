// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component renders correctly 1`] = `
<div>
  <div
    className="divFlex"
    data-testid="SearchBar"
  >
    <input
      className="inputFlex"
      onChange={[Function]}
      placeholder="Search by title..."
      type="text"
    />
  </div>
</div>
`;
